### PR TITLE
#111 renderItem is sometimes called with undefined

### PR DESCRIPTION
**Pull request by joekukish**
* Checked against the amount of items instead of assuming it is the same collection.

### CODE REVIEW

File: packages/lit-virtualizer/src/LitVirtualizer.ts [hunk 1]
```diff
@@ -113,11 +113,14 @@ export class LitVirtualizer extends LitElement {
     render(): TemplateResult {
         const { items, _renderItem, keyFunction } = this;
         const itemsToRender = [];
+        const lastItem = Math.min(items.length, this._last + 1);
+
         if (this._first >= 0 && this._last >= this._first) {
-            for (let i = this._first; i < this._last + 1; i++) {
+            for (let i = this._first; i < lastItem; i++) {
```
Comment (justinfagnani) [inline packages/lit-virtualizer/src/LitVirtualizer.ts:119]: I think this needs to be fixed here too: https://github.com/PolymerLabs/uni-virtualizer/blob/d90cbcc4430abf45abba007105ae2ec482eefbde/packages/lit-virtualizer/src/virtualize.ts#L66

Comment (graynorton) [reply packages/lit-virtualizer/src/LitVirtualizer.ts:119]: Yes—the directive and the element used to share a single implementation and we may be able to get back to that state eventually, but for now this logic needs to be maintained in both places.

**Review by graynorton — CHANGES_REQUESTED** [review]
Thanks for the PR, @joekukish! I confirmed that this indeed fixes a reproducible bug, so we'll go ahead and merge as soon as you've applied to the directive implementation as well.

Longer term, I may want to satisfy ourselves that there isn't a better spot to fix the bug, as I'm not sure we should ever be getting to this code in a state where `_last` is greater than the  highest index in the array—but that will require a deeper dive than we have time for right now.

**Review by graynorton — APPROVED** [review]
In the interest of time, I'm going to go ahead and merge this; will then apply the same fix to the directive implementation.

**Comment by graynorton** [comment]
@joekukish Although I've merged this, I'm hoping you can help me with a simple repro case of your issue, ideally in the Lit playground.

I thought I'd made one the other day, but Chrome crashed and I lost my work. Frustratingly, when I tried to recreate it just now, I was no longer able to get the error to occur.

Here's my latest attempt: https://tinyurl.com/4kccv8zx

The steps I tried in pursuit of a reproduction:

1. Scroll past Item 50.
2. Click the `Toggle` button.

At this point, I would expect to see the error, since the previous value of `_last` is now outside the bounds of the new `items` array, but instead the list successfully re-renders and resets the scroll position to a point that's in bounds for the smaller data array.

If you could try to modify this example to show the problem you've experienced, that would be much appreciated!

**Comment by graynorton** [comment]
Actually, ignore my most recent comment. Here's a simple repro case: https://tinyurl.com/yes8b2e6

**Comment by joekukish** [comment]
@graynorton I'm glad you could reproduce it.

**Pull request merged**